### PR TITLE
chore: change to use uds-common test action

### PR DESCRIPTION
**Pull request by ericwyles**
## Description
Updated the test workflow to use new test action from uds-common

Updated other uds-common references to the latest version.

Refactoring workflow names and layout to match other swf packages.

## Related Issue

Relates to https://github.com/defenseunicorns/uds-software-factory/issues/17

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-mattermost/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .github/workflows/tag-and-release.yaml [hunk 1]
```diff
@@ -11,21 +11,48 @@ permissions:
...
     permissions: write-all
     runs-on: ubuntu-latest
     outputs:
       release_created: ${{ steps.release-flag.outputs.release_created }}
     steps:
-      - name: Create Release Tag
+      - name: Create release tag
         id: tag
         uses: google-github-actions/release-please-action@cc61a07e2da466bebbc19b3a7dd01d6aecb20d1e # v4.0.2
         with:
```
Comment (Racer159) [inline .github/workflows/tag-and-release.yaml:23]: This with isn't in the others.

File: .github/workflows/tag-and-release.yaml [hunk 2]
```diff
@@ -11,21 +11,48 @@ permissions:
...
+    
+      - name: Login to GHCR
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3
+        with:
+          registry: ghcr.io
+          username: dummy
+          password: ${{ secrets.GITHUB_TOKEN }}
+    
+      - name: Publish Package
+        run: uds run -f tasks/publish.yaml package --set FLAVOR=${{ matrix.flavor }}
```
Comment (Racer159) [inline .github/workflows/tag-and-release.yaml:58]: Save Logs

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -22,29 +22,22 @@ jobs:
         flavor: [upstream, registry1]
```
Comment (Racer159) [inline .github/workflows/test.yaml:22]: This should have the paths-ignore on it as well

**Review by Racer159 — CHANGES_REQUESTED** [review]
we should also update to the common commit lint action

File: .github/workflows/tag-and-release.yaml (hunk 1 above)
Comment (ericwyles) [reply .github/workflows/tag-and-release.yaml:23]: Thanks for catching that. I had to look up what it even did, and it's redundant now. The option we were passing the in 'with' is now the default behavior for release-please so I am removing it to be consistent.

File: .github/workflows/test.yaml (hunk 3 above)
Comment (ericwyles) [reply .github/workflows/test.yaml:39]: Updated.

File: .github/workflows/tag-and-release.yaml (hunk 2 above)
Comment (ericwyles) [reply .github/workflows/tag-and-release.yaml:56]: Updated.

**Review by zachariahmiller — APPROVED** [review]

**Review by Racer159 — APPROVED** [review]

**Pull request merged**